### PR TITLE
fix bug when master worker delete, the dataset and runtime status do …

### PR DESCRIPTION
**Pull request by hahchenchen**
…not changed.

Signed-off-by: sichen.zhao <sichen.zhao@outlook.com>

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by hahchenchen** [comment]
fix the bug #2389 2389

File: pkg/ddc/goosefs/health_check.go [hunk 1]
```diff
@@ -324,3 +325,9 @@ func (e *GooseFSEngine) checkFuseHealthy() (err error) {
 	}
 	return err
 }
+
+
+// CheckExistenceOfEngine check the engine existed
+func (e *GooseFSEngine) CheckExistenceOfEngine() (err error) {
+	return
+}
```
Comment (cheyang) [inline pkg/ddc/goosefs/health_check.go:333]: Please add return of line. Thanks.

Comment (hahchenchen) [reply pkg/ddc/goosefs/health_check.go]: done

File: pkg/ddc/alluxio/health_check.go [hunk 2]
```diff
@@ -322,3 +323,101 @@ func (e *AlluxioEngine) checkFuseHealthy() (err error) {
 	}
 	return err
 }
+
+
+// CheckExistenceOfEngine check engine existed
+func (e *AlluxioEngine) CheckExistenceOfEngine() (err error) {
+
+	master, masterErr := kubeclient.GetStatefulSet(e.Client, e.getMasterName(), e.namespace)
```
Comment (cheyang) [inline pkg/ddc/alluxio/health_check.go:331]: I'm wondering the reason you don't put this logic into checkMasterHealthy, checkWorkerHealthy, checkFuseHealthy separately?

**Review by cheyang — APPROVED** [review]

**Pull request merged**